flink-core/src/main/java/org/apache/flink/api/common/typeutils/CompositeTypeSerializerConfigSnapshot.java

```diff
@@ -73,13 +91,11 @@ public boolean equals(Object obj) {
 		}
 
 		return (obj.getClass().equals(getClass()))
-				&& Arrays.equals(
-					nestedSerializerConfigSnapshots,
-					((CompositeTypeSerializerConfigSnapshot) obj).getNestedSerializerConfigSnapshots());
+				&& nestedSerializersWithConfigs.equals(((CompositeTypeSerializerConfigSnapshot) obj).getNestedSerializersAndConfigs());
 	}
 
 	@Override
 	public int hashCode() {
-		return Arrays.hashCode(nestedSerializerConfigSnapshots);
+		return nestedSerializersWithConfigs.hashCode();
```

Comment:
Same here; `nestedSerializersWithConfigs` is a `List` not an array, the `hashCode` implementation should suffice?